httplib.h

```diff
@@ -4409,8 +4414,69 @@ inline bool load_system_certs_on_windows(X509_STORE *store) {
 
   return true;
 }
+#elif defined(__APPLE__)
+using CFObjectDeleter = std::function<void(CFTypeRef)>;
+template <typename T> using CFObjectPtr = std::unique_ptr<T, CFObjectDeleter>;
+
+inline bool load_system_certs_on_apple(X509_STORE *store) {
+  CFStringRef keys[] = {kSecClass, kSecMatchLimit, kSecReturnRef};
+  CFTypeRef values[] = {kSecClassCertificate, kSecMatchLimitAll,
+                        kCFBooleanTrue};
+
+  auto deleter = [](CFTypeRef p) {
+    if (p) CFRelease(p);
+  };
```

Comment:
A simple function reference should suffice, there's no need to use `std::function`:

```suggestion
static void CFObjectDeleter(CFTypeRef p) {
  if (p) CFRelease(p);
}
template <typename T> using CFObjectPtr = std::unique_ptr<T, decltype(CFObjectDeleter)*>;

inline bool load_system_certs_on_apple(X509_STORE *store) {
  CFStringRef keys[] = {kSecClass, kSecMatchLimit, kSecReturnRef};
  CFTypeRef values[] = {kSecClassCertificate, kSecMatchLimitAll,
                        kCFBooleanTrue};
```

Additionally, since the `p` is always checked and used before destruction, it cannot possibly be `NULL`, so you could drop the `CFObjectDeleter` function entirely and directly pass `CFRelease` to the `decltype()` call.

Reply:
I would prefer to keep the lambda, but I agree that we can not use std::function. A simple func ptr is enough.

Reply:
Yeah, a lambda is cleaner, well done :)